src/nvidia_rag/rag_server/vlm.py

```
@@ -301,70 +291,186 @@ def extract_and_process_messages(
+                })
+                if remaining_image_budget is not None:
+                    remaining_image_budget -= 1
+            except Exception:
+                continue
+        return parts
 
-        citations_instruct_user_message = HumanMessage(content=content_parts)
+    def _build_content_parts_by_page(
+        self,
+        vlm_template: dict[str, Any],
+        textual_context: str,
```

Ruff (0.15.0)

[warning] Unused method argument: `textual_context`

(ARG002)